retrieval/scrape_test.go

```diff
@@ -875,11 +876,11 @@ type errorAppender struct {
 func (app *errorAppender) Add(lset labels.Labels, t int64, v float64) (string, error) {
 	switch lset.Get(model.MetricNameLabel) {
 	case "out_of_order":
-		return "", storage.ErrOutOfOrderSample
+		return "", tsdb.ErrOutOfOrderSample
 	case "amend":
-		return "", storage.ErrDuplicateSampleForTimestamp
+		return "", tsdb.ErrAmendSample
 	case "out_of_bounds":
-		return "", storage.ErrOutOfBounds
+		return "", tsdb.ErrOutOfBounds
```

Comment:
But they are literally defined as the equality. This appender simulates a `storage.Appender` and not a `tsdb.Appender`.

Thinking about it. The change to `interface.go` actually makes package storage depend on tsdb and thereby all packages using a storage depend on tsdb.
This kind of breaks the entire purpose of the shim layer. So either we keep it separated as before and do explicit conversion in the shim layers or we move away from it and make dependencies to tsdb tighter.